QRCoder/QRCodeData.cs

```diff
@@ -59,42 +59,44 @@ public QRCodeData(string pathToRawData, Compression compressMode) : this(File.Re
     /// <param name="compressMode">The compression mode used for the raw data.</param>
     public QRCodeData(byte[] rawData, Compression compressMode)
     {
-        var bytes = new List<byte>(rawData);
-
         //Decompress
         if (compressMode == Compression.Deflate)
         {
-            using var input = new MemoryStream(bytes.ToArray());
+            using var input = new MemoryStream(rawData);
             using var output = new MemoryStream();
             using (var dstream = new DeflateStream(input, CompressionMode.Decompress))
             {
                 dstream.CopyTo(output);
             }
-            bytes = new List<byte>(output.ToArray());
+            rawData = output.ToArray();
         }
         else if (compressMode == Compression.GZip)
         {
-            using var input = new MemoryStream(bytes.ToArray());
+            using var input = new MemoryStream(rawData);
             using var output = new MemoryStream();
             using (var dstream = new GZipStream(input, CompressionMode.Decompress))
             {
                 dstream.CopyTo(output);
             }
-            bytes = new List<byte>(output.ToArray());
+            rawData = output.ToArray();
         }
 
-        if (bytes[0] != 0x51 || bytes[1] != 0x52 || bytes[2] != 0x52)
+        var count = rawData.Length;
+
+        if (count < 5)
```

Comment:
```suggestion
        if (rawData.Length < 5)
```
To give the JIT some hints, so the bound checks in the array access below can be removed (at least on newer JIT versions), and there's no harm for older ones.